### PR TITLE
🟡(change): cache organization response

### PR DESCRIPTION
**Pull request by ruancomelli**
Organization is pulled quite a bit from the application, but should not be updated often.

## Summary by Sourcery

This pull request introduces caching for organization data using Redis, removes the rainbow animation feature from the theme provider, and consolidates organization-related API endpoints and hooks. Additionally, it updates dependencies to include the new Redis client library.

- **New Features**:
    - Introduced caching for organization data using Redis to reduce database load and improve response times.
- **Enhancements**:
    - Removed the rainbow animation feature from the theme provider and related components.
    - Consolidated organization-related API endpoints and hooks for better consistency and maintainability.
- **Chores**:
    - Updated dependencies in go.mod to include new Redis client library.

### CODE REVIEW

**Comment by ruancomelli** [comment]
This is an experiment review for experiment `improve_review_guide`
Run ID: `improve_review_guide/run_2024-08-12T00-55-41_v1-21-0-74-g8aa259cc3-dirty`
The benchmark review for this pull request can be found at https://github.com/sourcery-ai-experiments/Trenova/pull/19.
This pull request was cloned from `https://github.com/sourcery-ai-experiments/Trenova/pull/11`. _(Note: the URL is not a link to avoid triggering a notification on the original pull request.)_

<details>
<summary>Experiment configuration</summary>

```yaml
review_config:
  # User configuration for the review
  # - benchmark - use the user config from the benchmark reviews
  # - <value> - use the value directly
  user_review_config:
    enable_ai_review: true
    enable_rule_comments: false

    enable_complexity_comments: false
    enable_security_comments: false
    enable_tests_comments: false
    enable_comment_suggestions: false

    enable_claude_comments: true
    enable_functionality_review: false

    enable_pull_request_summary: false
    enable_review_guide: true

    enable_approvals: false

  ai_review_config:
    # The model responses to use for the experiment
    # - benchmark - use the model responses from the benchmark reviews
    # - llm - call the language model to generate responses
    model_responses:
      comments_model: benchmark
      comment_area_model: benchmark
      comment_validation_model: benchmark
      comment_suggestion_model: benchmark
      complexity_model: benchmark
      docstrings_model: benchmark
      functionality_model: benchmark
      overall_comments_model: benchmark
      security_model: benchmark
      tests_model: benchmark
      pull_request_summary_model: benchmark
      review_guide_model: llm

# The pull request dataset to run the experiment on
pull_request_dataset:
  # open-source PRs
- https://github.com/sourcery-ai-experiments/gpt-crawler/pull/1
- https://github.com/sourcery-ai-experiments/graphviz2drawio/pull/1
- https://github.com/sourcery-ai-experiments/graphviz2drawio/pull/2
- https://github.com/sourcery-ai-experiments/iptux/pull/5
- https://github.com/sourcery-ai-experiments/iptux/pull/6
- https://github.com/sourcery-ai-experiments/kalima/pull/3
- https://github.com/sourcery-ai-experiments/pydantic_tfl_api/pull/1
- https://github.com/sourcery-ai-experiments/pydantic_tfl_api/pull/2
- https://github.com/sourcery-ai-experiments/Trenova/pull/11
  # Sourcery PRs
- https://github.com/sourcery-ai/core/pull/4055
- https://github.com/sourcery-ai/core/pull/4263
- https://github.com/sourcery-ai/core/pull/4352
- https://github.com/sourcery-ai/core/pull/4364
- https://github.com/sourcery-ai/core/pull/4366
- https://github.com/sourcery-ai/core/pull/4390
- https://github.com/sourcery-ai/core/pull/4402
- https://github.com/sourcery-ai/core/pull/4445
- https://github.com/sourcery-ai/core/pull/4462
- https://github.com/sourcery-ai/core/pull/4463

review_comment_labels: []

# Benchmark reviews generated by running
#   python -m scripts.experiment benchmark <experiment_name>
benchmark_reviews:
- dataset_pull_request: https://github.com/sourcery-ai/core/pull/4463
  review_pull_request: https://github.com/sourcery-ai-experiments/core/pull/236
- dataset_pull_request: https://github.com/sourcery-ai-experiments/gpt-crawler/pull/1
  review_pull_request: https://github.com/sourcery-ai-experiments/gpt-crawler/pull/10
- dataset_pull_request: https://github.com/sourcery-ai-experiments/iptux/pull/5
  review_pull_request: https://github.com/sourcery-ai-experiments/iptux/pull/32
- dataset_pull_request: https://github.com/sourcery-ai-experiments/iptux/pull/6
  review_pull_request: https://github.com/sourcery-ai-experiments/iptux/pull/33
- dataset_pull_request: https://github.com/sourcery-ai-experiments/kalima/pull/3
  review_pull_request: https://github.com/sourcery-ai-experiments/kalima/pull/17
- dataset_pull_request: https://github.com/sourcery-ai/core/pull/4055
  review_pull_request: https://github.com/sourcery-ai-experiments/core/pull/258
- dataset_pull_request: https://github.com/sourcery-ai/core/pull/4263
  review_pull_request: https://github.com/sourcery-ai-experiments/core/pull/259
- dataset_pull_request: https://github.com/sourcery-ai/core/pull/4352
  review_pull_request: https://github.com/sourcery-ai-experiments/core/pull/260
- dataset_pull_request: https://github.com/sourcery-ai/core/pull/4364
  review_pull_request: https://github.com/sourcery-ai-experiments/core/pull/261
- dataset_pull_request: https://github.com/sourcery-ai/core/pull/4366
  review_pull_request: https://github.com/sourcery-ai-experiments/core/pull/262
- dataset_pull_request: https://github.com/sourcery-ai/core/pull/4390
  review_pull_request: https://github.com/sourcery-ai-experiments/core/pull/263
- dataset_pull_request: https://github.com/sourcery-ai/core/pull/4402
  review_pull_request: https://github.com/sourcery-ai-experiments/core/pull/264
- dataset_pull_request: https://github.com/sourcery-ai/core/pull/4445
  review_pull_request: https://github.com/sourcery-ai-experiments/core/pull/265
- dataset_pull_request: https://github.com/sourcery-ai/core/pull/4462
  review_pull_request: https://github.com/sourcery-ai-experiments/core/pull/266
- dataset_pull_request: https://github.com/sourcery-ai-experiments/graphviz2drawio/pull/1
  review_pull_request: https://github.com/sourcery-ai-experiments/graphviz2drawio/pull/23
- dataset_pull_request: https://github.com/sourcery-ai-experiments/graphviz2drawio/pull/2
  review_pull_request: https://github.com/sourcery-ai-experiments/graphviz2drawio/pull/24
- dataset_pull_request: https://github.com/sourcery-ai-experiments/pydantic_tfl_api/pull/1
  review_pull_request: https://github.com/sourcery-ai-experiments/pydantic_tfl_api/pull/19
- dataset_pull_request: https://github.com/sourcery-ai-experiments/pydantic_tfl_api/pull/2
  review_pull_request: https://github.com/sourcery-ai-experiments/pydantic_tfl_api/pull/20
- dataset_pull_request: https://github.com/sourcery-ai-experiments/Trenova/pull/11
  review_pull_request: https://github.com/sourcery-ai-experiments/Trenova/pull/19

```

</details>

**Comment by SourceryAI** [comment]
## Reviewer's Guide by Sourcery

This pull request introduces caching for organization data using Redis, removes the rainbow animation feature, and consolidates organization-related API endpoints and hooks. The main changes include:

1. Implementation of Redis caching for organization data to improve performance.
2. Removal of the rainbow animation feature from the theme provider and related components.
3. Consolidation of organization-related API endpoints and hooks for better consistency.
4. Addition of a new Redis client package and integration with the server.
5. Updates to the organization service to use the new caching mechanism.
6. Modifications to frontend components and hooks to align with the backend changes.

The changes aim to enhance performance through caching and streamline the codebase by removing unused features and consolidating related functionality.

### File-Level Changes

| Files | Changes |
| ----- | ------- |
| `internal/api/services/organization.service.go` | Implemented Redis caching for organization data in the OrganizationService |
| `web/frontend/src/components/ui/theme-provider.tsx`<br/>`web/frontend/src/components/layout/user-avatar-menu.tsx`<br/>`web/frontend/src/components/layout/topbar.tsx` | Removed rainbow animation feature from theme provider and related components |
| `internal/api/handlers/organization.handler.go`<br/>`web/frontend/src/hooks/useQueries.ts`<br/>`web/frontend/src/services/OrganizationRequestService.ts` | Consolidated organization-related API endpoints and hooks |
| `pkg/redis/client.go`<br/>`internal/server/server.go`<br/>`cmd/serve.go`<br/>`config/server_config.go` | Added new Redis client package and integrated it with the server |
| `pkg/models/organization.go` | Updated organization model to support caching |
| `web/frontend/src/components/layout/logo.tsx`<br/>`web/frontend/src/pages/admin/Dashboard.tsx` | Modified frontend components to use updated organization hooks |
| `go.mod` | Updated dependencies in go.mod to include new Redis client library |

---

<details>
<summary>Tips</summary>

- Trigger a new Sourcery review by commenting `@sourcery-ai review` on the pull request.
- Continue your discussion with Sourcery by replying directly to review comments.
- You can change your review settings at any time by accessing your [dashboard](https://app.sourcery.ai):
    - Enable or disable the Sourcery-generated pull request summary or reviewer's guide;
    - Change the review language;
- You can always [contact us](mailto:support@sourcery.ai) if you have any questions or feedback.

</details>

File: internal/api/services/organization.service.go [hunk 1]
```diff
@@ -40,30 +43,36 @@ func NewOrganizationService(s *server.Server) *OrganizationService {
...
+	return fmt.Sprintf("organization:%s", orgID.String())
+}
 
-	err := s.db.NewSelect().
-		Model(org).
-		Relation("State").
-		Where("organization.business_unit_id = ?", buID).
-		Where("organization.id = ?", orgID).
-		Scan(ctx)
+func (s *OrganizationService) GetOrganization(ctx context.Context, buID, orgID uuid.UUID) (*models.Organization, error) {
```
Comment (SourceryAI) [inline internal/api/services/organization.service.go:55]: **suggestion:** Simplify error handling for cache fetch and unmarshal.

The error handling for fetching and unmarshalling the organization from the cache is verbose. Consider combining the error checks to simplify the code and improve readability.

**Review by SourceryAI — COMMENTED** [review]
Hey @ruancomelli - I've reviewed your changes - here's some feedback:

**Overall Comments**:
- Consider encapsulating repetitive error-handling patterns for cache operations into utility functions to keep the code cleaner.
- The removal of the rainbow animation feature seems to be a separate concern from the caching implementation. It would be better to split these changes into separate pull requests.
- There's a minor typo in the comment: 'Try toi fetch the organization from the cache.' It should be 'Try to fetch the organization from the cache.'

<details>
<summary>Here's what I looked at during the review</summary>

- 🟡 **General issues**: 1 issue found
- 🟢 **Documentation**: all looks good
</details>

***

[LangSmith trace](https://smith.langchain.com/o/7d8518ad-6dd3-5c7b-96d2-757769296335/projects/p/88560225-5072-48bc-967c-dc54da53f265/r/9394016c-6b2a-48ea-9215-ec382980aeb6?poll=true)

<sub>
Help me be more useful! Please click 👍 or 👎 on each comment to tell me if it was helpful.
</sub>

**Pull request closed without merging**